docs/contracts.rst

```diff
@@ -284,8 +284,38 @@ It will generate a function of the following form::
         b = data[arg1][arg2][arg3].b;
     }
 
-Note that the mapping in the struct is omitted because there
-is no good way to provide the key for the mapping.
+    .. note::
```

Comment:
Perhaps it would be better to move the complex example abov at the end?

Reply:
@chriseth I feel like it makes more sense here, as it shows an example of arrays with single elements and then leads to the whole array example. Unless I'm missing something?

Reply:
I mean the complex example that involves mappings and stuff.

Reply:
@chriseth I'm still not 100% certain, but I rearranged it.